test/src/test/java/jenkins/SkipSomeTests.java

```diff
@@ -0,0 +1,40 @@
+package jenkins;
+
+/**
+ * @author MarkEWaite
+ */
+public class SkipSomeTests {
+
+    private static final String JENKINS_URL = System.getenv("JENKINS_URL") != null
+            ? System.getenv("JENKINS_URL")
+            : "http://localhost:8080/";
+
+    private static final String GIT_BRANCH = System.getenv("GIT_BRANCH") != null
+            ? System.getenv("GIT_BRANCH")
+        : "GIT_BRANCH environment variable is null"; // intentionally not a typical branch name
+
+    public enum ReasonTestShouldRun {
+        NEVER_FAILING_TEST, /* Test has not failed in a very long time, run it less often */
+    }
+
+    /**
+     * Run a test sometimes, based on the reason the test should run.
+     * Reduce costs by running tests less frequently when they use
+     * this method in an Assume expression.
+     */
+    public static boolean runTestSometimes(ReasonTestShouldRun testReason) {
+        switch (testReason) {
+        case NEVER_FAILING_TEST:
+            if (JENKINS_URL.equals("https://ci.jenkins.io/")) {
+                if (GIT_BRANCH.equals("master") || GIT_BRANCH.equals("origin/master")) {
+                    /* Run 1 of 8 times on the master branch, typically once a week or more */
+                    /* The master branch builds complete 8 times per week (on average), with each build running 3 configurations */
+                    return (System.currentTimeMillis() % 8) == 0;
```

Comment:
how does this assure 1 in 8 times?

Wouldn't it be better to do off of the build number rather than wall clock time which could be completely random and never be mod 8?

Reply:
> how does this assure 1 in 8 times?

It doesn't assure 1 in 8 times, but it does approximate it.

When a change follows another change before the build has completed for the first change, the build for the first change is aborted.  By relying on the semi-random nature of the system clock, we're less likely to skip a build because it was aborted by a later merge.

By using the system clock and its expected relatively even distribution of values, when a test is about to be run, roughly 1 time out of 8 times it will be selected to run.  I think that is better than relying on the build number because it does not risk skipping a build because the build was aborted due to a later build.